### PR TITLE
Display hearing request type column on assign hearings table

### LINKED ISSUE #15059: Identify Virtual and former Travel Board hearings in scheduling queue
## User stories
As a hearing coordinator, I want to know the hearing type when I'm scheduling so I can know whether it’s ready to be scheduled while ROs are closed.

As a hearing coordinator, I want to know when a hearing used to be a Travel Board hearing so it can be properly prioritized.

## Background
As hearing coordinators begin to schedule directly to virtual, they'll need to know the hearing request type so they can see which are virtual and therefore ready to schedule right now.

## Acceptance criteria
- [ ] Cache data for formerly Travel Board
- [ ] Return data in format so column can display comma-separated values
- [ ] Remove Types column
- [ ] Replace with Appeal Stream Type and Hearing Type column
- [ ] Hearing Type column shows "Video" or "Virtual" 
- [ ] If a hearing was converted from Travel Board to virtual, the Hearing Type column also shows "former Travel"
- [ ] Multiple types are separated by commas 
- [ ] Keep State of Residence, move to end with Suggested Location column

## Out of scope
Sorting and filtering the Hearing Type column (#15203, #15058)

## Designs
Design task: #15057
Figma link: https://www.figma.com/proto/V87TZArfdurCGJiEjQ73ES/Virtual-Hearings?node-id=7050%3A18&scaling=min-zoom

## Release notes
TK

### PR DESCRIPTION
**Pull request by rubaiyat22**
Follows #15294
Resolves #15059

⚠️ This can only be deployed after #15294 has been merged and deployed to production

### Description
- create a new column `Hearing Type` on assign hearings table
- rename `Types` column to `Appeal Stream Type`
- break out tests from `spec/feature/hearings/schedule_veteran/build_hearsched_spec.rb` to a new test file `spec/feature/hearings/assign_hearings_table_spec.rb` and write some new tests
- fix the docket line display

### Testing Plan
Console
- run `UpdateCachedAppealsAttributesJob.new.perform_now`

Steps
- login as `BVASYELLOW` and go to `http://localhost:3000/hearings/schedule/assign`
- choose a RO which has veterans waiting
- look at `Legacy Veterans Waiting` and `AMA Veterans Waiting`

### User Facing Changes
 - [x] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---
Screen Shot 2020-09-24 at 4 26 50 PM |Screen Shot 2020-09-24 at 4 25 50 PM |

### CODE REVIEW

File: spec/feature/hearings/assign_hearings_table_spec.rb [hunk 1]
```diff
@@ -0,0 +1,448 @@
+# frozen_string_literal: true
```
Comment (rubaiyat22) [inline spec/feature/hearings/assign_hearings_table_spec.rb:1]: most of this file was ported over from `spec/feature/hearings/schedule_veteran/build_hearsched_spec.rb` and left untouched

File: spec/feature/hearings/assign_hearings_table_spec.rb [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+
+        step "clicking on a filter reduces the number of results by the expect amount" do
+          page.find("label", text: "#{Appeal.first.representative_name} (1)", match: :prefer_exact).click
+          expect(find("tbody").find_all("tr").length).to eq(1)
+        end
+      end
+    end
+  end
+
+  context "Hearing Request Type column" do
```
Comment (rubaiyat22) [inline spec/feature/hearings/assign_hearings_table_spec.rb:375]: new tests from here down

File: spec/feature/hearings/schedule_veteran/build_hearsched_spec.rb [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+##
```
Comment (rubaiyat22) [inline spec/feature/hearings/schedule_veteran/build_hearsched_spec.rb:3]: Only change in this file was to port some tests over to `spec/feature/hearings/assign_hearings_table_spec.rb`

File: app/controllers/hearings/schedule_hearing_tasks_controller.rb [hunk 4]
```diff
@@ -17,14 +17,20 @@ def index
       regional_office_key: allowed_params[:regional_office_key]
     )
 
-    # Select `power_of_attorney_name` from the `CachedAppeal` table. This is an
+    # Select `power_of_attorney_name`, `hearing_request_type`, and
+    # `formally_travel` from the `CachedAppeal` table. This is an
     # optimization that fetches the POA name from the `CachedAppeal` table,
     # where it is already cached for filtering. The `TaskColumnSerializer` is
     # "aware" of this optimization, and will serialize the cached value instead
     # of getting the value through the `Appeal` instance.
```
Comment (tomas-nava) [inline app/controllers/hearings/schedule_hearing_tasks_controller.rb:25]:
```suggestion
    # optimization that fetches these values from the `CachedAppeal` table,
    # where they are already cached for filtering. The `TaskColumnSerializer` is
    # "aware" of these optimizations, and will serialize the cached values instead
    # of getting them through the `Appeal` instance.
```

File: client/app/hearings/components/assignHearings/AssignHearingsFields.jsx [hunk 5]
```diff
@@ -119,3 +119,20 @@ SuggestedHearingLocation.propTypes = {
...
 };
+
+export const HearingRequestType = ({ hearingRequestType, isFormallyTravel }) => {
+  if (!hearingRequestType) {
+    return null;
+  }
+
+  return (
+    <React.Fragment>
+      {isFormallyTravel ? <span>{`formally Travel, ${hearingRequestType}`}</span> : <span>{hearingRequestType}</span>}
```
Comment (tomas-nava) [inline client/app/hearings/components/assignHearings/AssignHearingsFields.jsx:130]:
```suggestion
      {isFormallyTravel ? <span>{`former Travel, ${hearingRequestType}`}</span> : <span>{hearingRequestType}</span>}
```
[the design](https://www.figma.com/proto/V87TZArfdurCGJiEjQ73ES/Virtual-Hearings?node-id=7050%3A20&scaling=min-zoom) shows this as "former Travel"

File: client/app/hearings/components/assignHearings/AssignHearingsTable.jsx [hunk 6]
```diff
@@ -113,13 +114,24 @@ export default class AssignHearingsTable extends React.PureComponent {
...
+        name: 'hearingRequestType',
+        header: 'Hearing Type',
+        align: 'left',
+        valueFunction: (row) => (
+          <HearingRequestType
+            hearingRequestType={row.hearingRequestType}
+            isFormallyTravel={row.isFormallyTravel}
+          />
+        )
+      },
```
Comment (tomas-nava) [inline client/app/hearings/components/assignHearings/AssignHearingsTable.jsx:134]: I like how you made this its own component.

File: spec/feature/hearings/assign_hearings_table_spec.rb (hunk 2 above)
Comment (tomas-nava) [reply spec/feature/hearings/assign_hearings_table_spec.rb:375]: could we have a simple test to validate that the columns display as expected for an AMA appeal as well?

**Review by tomas-nava — CHANGES_REQUESTED** [review]
This looks excellent!

I made a few suggestions, only one of which is blocking (show **"former Travel"** in the column instead of **"formally Travel"**). 

Related to that, though, I think we should replace **"formally"** everywhere with **"former"** or **"formerly"**, to be consistent with what we're displaying to the user. Unfortunately that includes columns in the `cached_appeal_attributes` table.

File: spec/feature/hearings/assign_hearings_table_spec.rb (hunk 2 above)
Comment (rubaiyat22) [reply spec/feature/hearings/assign_hearings_table_spec.rb:375]: Done!

**Review by tomas-nava — APPROVED** [review]
approve!

**Pull request merged**